app/models/manageiq/providers/vmware/infra_manager/refresh_worker/runner.rb

```diff
@@ -33,11 +48,20 @@ def start_inventory_collector(collector)
     end
 
     _log.info("Started inventory collector")
+  end
 
-    thread
+  def ensure_inventory_collector
+    return if collector_thread&.alive?
+
+    _log.warn("Inventory collector thread not running, restarting...")
+    start_inventory_collector
   end
 
-  def stop_inventory_collector(collector)
+  def stop_inventory_collector
     collector.stop
   end
+
+  def update_driven_refresh?
```

Comment:
we should probably move this into ems later, so we can use this on other places?

Reply:
:+1: agreed